desktop/AutoStartManager.ts

```diff
@@ -51,21 +57,72 @@ class AutoStartManager {
       path: isLinuxAppImage() ? process.env.APPIMAGE : process.execPath,
     });
 
-    ipcMain.removeAllListeners(ipcConsts.TOGGLE_AUTO_START);
-    ipcMain.handle(ipcConsts.TOGGLE_AUTO_START, () =>
-      AutoStartManager.toggleAutoStart()
-    );
-
-    ipcMain.removeHandler(ipcConsts.IS_AUTO_START_ENABLED_REQUEST);
-    ipcMain.handle(ipcConsts.IS_AUTO_START_ENABLED_REQUEST, () =>
-      AutoStartManager.isEnabled()
-    );
+    const ipcEventMappings = [
+      {
+        event: ipcConsts.SYNC_IS_AUTO_START_ON_SYSTEM_LAUNCH_REQUEST,
+        handler: AutoStartManager.syncIsAutoStartOnLoginEnabled,
+      },
+      {
+        event: ipcConsts.TOGGLE_AUTO_START_ON_SYSTEM_LAUNCH,
+        handler: AutoStartManager.toggleAutoStart,
+      },
+      {
+        event: ipcConsts.IS_AUTO_START_ON_SYSTEM_LAUNCH_ENABLED_REQUEST,
+        handler: AutoStartManager.isEnabled,
+      },
+    ];
+
+    ipcEventMappings.forEach(({ event, handler }) => {
+      ipcMain.removeHandler(event);
+      ipcMain.handle(event, handler);
+    });
   }
 
-  static isEnabled = () => AutoStartManager.service.isEnabled();
+  static isEnabled = () =>
+    StoreService.get(IS_AUTO_START_ENABLED) ??
+    StoreService.get(IS_AUTO_START_ON_SYSTEM_LAUNCH_ENABLED);
+
+  static syncIsAutoStartOnLoginEnabled = async () => {
+    try {
+      // call the function only once on launch
+      if (AutoStartManager.isSyncFinished) {
+        return { status: true };
+      }
+
+      AutoStartManager.isSyncFinished = true;
+
+      const isEnabled = await AutoStartManager.service.isEnabled();
+
+      // migration process with auto deletion the flag from
+      if (StoreService.has(IS_AUTO_START_ENABLED)) {
```

Comment:
Shouldn't it check for `isEnabled` variable instead? 🤔

Reply:
We rely on the config value if we have a different one in the system. Based on the config file, we synchronize the state to maintain the correct value. During migration, we sync the isEnabled() function to migrate the value without resetting the current state.